cmd/csi-attacher/main.go

```diff
@@ -280,11 +296,37 @@ func main() {
 		supportsListVolumesPublishedNodes,
 		*reconcileSync,
 	)
+	// handle SIGTERM and SIGINT by cancelling the context.
+	ctx, terminate := context.WithCancel(ctx)
+	runCtx, cancelRun := context.WithCancel(ctx)
+	shutdownHandler := server.SetupSignalHandler()
+
+	defer terminate()
+
+	var signalReceived bool
+
+	go func() {
+		defer cancelRun()
+		<-shutdownHandler
+		signalReceived = true
+		logger.Info("Received SIGTERM or SIGINT signal, shutting down controller.")
+	}()
```

Comment:
All this should be behind the feature gate too. We don't want to change the existing behavior.